ci/gpu_test.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+# Copyright (c) 2018-2022, NVIDIA CORPORATION.
+##############################################
+# cuML GPU build and test script for CI      #
+##############################################
+
+set -e
+NUMARGS=$#
+ARGS=$*
+
+# Arg parsing function
+function hasArg {
+    (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
+}
+
+# Set path and build parallel level
+export PATH=/opt/conda/bin:/usr/local/cuda/bin:$PATH
+export PARALLEL_LEVEL=${PARALLEL_LEVEL:-8}
+export CONDA_ARTIFACT_PATH=${WORKSPACE}/ci/artifacts/cuml/cpu/.conda-bld/
+
+# Set home to the job's workspace
+export HOME=$WORKSPACE
+
+# Determine CUDA release version
+export CUDA_REL=${CUDA%.*}
+
+# Switch to project root; also root of repo checkout
+cd $WORKSPACE
+
+# Parse git describe
+export GIT_DESCRIBE_TAG=`git describe --tags`
+export MINOR_VERSION=`echo $GIT_DESCRIBE_TAG | grep -o -E '([0-9]+\.[0-9]+)'`
+unset GIT_DESCRIBE_TAG
+
+# ucx-py version
+export UCX_PY_VERSION='0.27.*'
+
+# configure numba threading library
+export NUMBA_THREADING_LAYER=workqueue
+
+################################################################################
+# SETUP - Check environment
+################################################################################
+
+gpuci_logger "Check environment"
+env
+
+gpuci_logger "Check GPU usage"
+nvidia-smi
+
+gpuci_logger "Activate conda env"
+. /opt/conda/etc/profile.d/conda.sh
+conda activate rapids
+
+gpuci_logger "Install dependencies"
+gpuci_mamba_retry install -c conda-forge -c rapidsai -c rapidsai-nightly -c nvidia \
+      "cudatoolkit=${CUDA_REL}" \
+      "cudf=${MINOR_VERSION}" \
+      "rmm=${MINOR_VERSION}" \
+      "libcumlprims=${MINOR_VERSION}" \
+      "libraft-headers=${MINOR_VERSION}" \
+      "libraft-distance=${MINOR_VERSION}" \
+      "libraft-nn=${MINOR_VERSION}" \
+      "pyraft=${MINOR_VERSION}" \
+      "dask-cudf=${MINOR_VERSION}" \
+      "dask-cuda=${MINOR_VERSION}" \
+      "ucx-py=${UCX_PY_VERSION}" \
+      "ucx-proc=*=gpu" \
+      "xgboost=1.6.0dev.rapidsai${MINOR_VERSION}" \
+      "rapids-build-env=${MINOR_VERSION}.*" \
+      "rapids-notebook-env=${MINOR_VERSION}.*" \
+      "shap>=0.37,<=0.39"
+
+if [ "$(arch)" = "x86_64" ]; then
+    gpuci_mamba_retry install -c conda-forge -c rapidsai -c rapidsai-nightly -c nvidia \
+        "rapids-doc-env=${MINOR_VERSION}.*"
+fi
+
+# https://docs.rapids.ai/maintainers/depmgmt/
+# gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
+# gpuci_mamba_retry install -y "your-pkg=1.0.0"
+
+gpuci_logger "Check compiler versions"
+python --version
+$CC --version
+$CXX --version
+
+gpuci_logger "Check conda environment"
+conda info
+conda config --show-sources
+conda list --show-channel-urls
+
+if hasArg --skip-tests; then
+    gpuci_logger "Skipping Tests"
+    exit 0
+fi
```

Comment:
if this script is only ever invoked via the Jenkinsfile without any arguments, then we can remove any `hasArg` functionality